swaywall

```diff
@@ -75,19 +85,21 @@ def main():
     hst_file = Path(state) / "wallpaperhst"
     img_exts = ["png", "jpg", "jpeg"]
 
-    hst = get_history(hst_file)
-    if args.restore and hst:
-        set_wall(hst[0])
-        exit()
-
-    walls = find(walls_dir, img_exts)
-    if not walls:
-        raise FileNotFoundError(f"no wallpapers found in {walls_dir}")
-
-    new = get_new(walls, hst)
-    if new:
-        remember(new, walls, hst, hst_file)
-        set_wall(new)
+    outputs = get_outputs()
+    for output in outputs:
+        hst = get_history(hst_file)
+        if args.restore and hst:
+            set_wall(hst[0], output)
+            exit()
+
+        walls = find(walls_dir, img_exts)
+        if not walls:
+            raise FileNotFoundError(f"no wallpapers found in {walls_dir}")
```

Comment:
This should not be inside the for loop (as otherwise, of course, these lookups will be unnecessarily performed as many times as there are outputs).